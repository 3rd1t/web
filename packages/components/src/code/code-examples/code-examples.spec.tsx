import React from "react"
import CodeExample from "./code-examples"
import { fireEvent, render } from "@testing-library/react"

describe("CodeExample", () => {
  const component = <CodeExample code={[{ language: "javascript", snippet: "const x = 1;" }]} />

  it("renders correctly", () => {
    const { baseElement } = render(component)
    expect(baseElement).toMatchSnapshot()
  })

  describe("with multiple languages", () => {
    describe("when the user clicks on a language button", () => {
      it("switches to the respective snippet", () => {
        const { getAllByTestId, queryByTestId } = render(
          <CodeExample
            code={[
              {
                language: "javascript",
                snippet: "const x = 1; ARBITRARY_TOKEN_IN_JAVASCRIPT",
              },
              {
                language: "python",
                snippet: "y = 2 ARBITRARY_TOKEN_IN_PYTHON",
              },
            ]}
          />,
        )

        const selectButtons = getAllByTestId("selectLanguage")

        expect(selectButtons.length).toBe(2)
        expect(selectButtons[0].innerHTML).toEqual("javascript")
        expect(selectButtons[1].innerHTML).toEqual("python")

        expect(queryByTestId("syntaxHighlighter").innerHTML.includes("ARBITRARY_TOKEN_IN_JAVASCRIPT")).toBe(true)
        expect(queryByTestId("syntaxHighlighter").innerHTML.includes("ARBITRARY_TOKEN_IN_PYTHON")).toBe(false)

        fireEvent.click(selectButtons[1])
        expect(queryByTestId("syntaxHighlighter").innerHTML.includes("ARBITRARY_TOKEN_IN_JAVASCRIPT")).toBe(false)
        expect(queryByTestId("syntaxHighlighter").innerHTML.includes("ARBITRARY_TOKEN_IN_PYTHON")).toBe(true)

        fireEvent.click(selectButtons[0])
        expect(queryByTestId("syntaxHighlighter").innerHTML.includes("ARBITRARY_TOKEN_IN_JAVASCRIPT")).toBe(true)
        expect(queryByTestId("syntaxHighlighter").innerHTML.includes("ARBITRARY_TOKEN_IN_PYTHON")).toBe(false)
      })
    })
  })
})
