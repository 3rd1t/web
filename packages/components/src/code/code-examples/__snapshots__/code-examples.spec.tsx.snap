// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CodeExample renders correctly 1`] = `
<body>
  <div>
    <div
      class="shadow-lg"
    >
      <nav
        class="flex items-center justify-start px-4 space-x-4 border border-gray-500 rounded-t"
      >
        <button
          class="py-3 mr-8 text-xs font-semibold tracking-wide no-underline uppercase focus:outline-none hover:text-gray-900 text-gray-900 "
          data-testid="selectLanguage"
        >
          javascript
        </button>
      </nav>
      <div
        class="px-4 overflow-x-visible overflow-y-hidden bg-gray-900 rounded-b"
      >
        <pre
          class="font-mono"
          data-testid="syntaxHighlighter"
          style="font-size: 14px; line-height: 1.375; direction: ltr; text-align: left; white-space: pre; word-spacing: normal; word-break: normal; tab-size: 4; hyphens: none; color: rgb(203, 213, 224); padding: 1em; margin: .5em 0px; overflow: auto;"
        >
          <code
            class="language-javascript"
            style="font-size: 14px; line-height: 1.375; direction: ltr; text-align: left; white-space: pre; word-spacing: normal; word-break: normal; tab-size: 4; hyphens: none; color: rgb(203, 213, 224);"
          >
            const x = 1;
          </code>
        </pre>
      </div>
    </div>
  </div>
</body>
`;
